src/libraries/System.Private.Uri/tests/FunctionalTests/UriTests.cs

```diff
@@ -818,5 +818,24 @@ public static void FilePathHandlesNonAscii(string uriString, string toString, st
             Assert.Equal(absoluteUri, uri2.AbsoluteUri);
             Assert.Equal(localPath, uri2.LocalPath);
         }
+
+        [Fact]
+        public static void ZeroPortIsParsedForBothKnownAndUnknownSchemes()
+        {
+            Uri.TryCreate("http://example.com:0", UriKind.Absolute, out var uri);
```

Comment:
Couldn't the test be rather parametrized instead of copying similar block 3 times in a row?